### PR TITLE
Feat/125 search

### PR DESCRIPTION
**Pull request by JDziewonski98**
# Summary

WIP of search panel
Accessible thru search button in top left corner
Murals are clickable, map will zoom to location
TODO: 
make each ~~mural~~ / tour / collection clickable for more info / zoom
add edit button beside each item if logged in as admin
styling
loading image for each result item

## Test Plan
Manually

## Related Issues
125

### CODE REVIEW

File: frontend/src/SearchMenu/SearchMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  const [murals, setMurals] = useState<any>([]);
+  const [tours, setTours] = useState<any>([]);
+  const [collections, setCollections] = useState<any>([]);
+  const [displayedMurals, setDisplayedMurals] = useState<any>([]);
+  const [displayedTours, setDisplayedTours] = useState<any>([]);
+  const [displayedCollections, setDisplayedCollections] = useState<any>([]);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+
+  /**
+   * fetch all collections/tours/murals upon component load LOL!
```
Comment (tedspare) [inline frontend/src/SearchMenu/SearchMenu.tsx:60]: 😂

File: frontend/src/SearchMenu/SearchMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    return collection.filter(function (item: any) {
+      return item.name.toLowerCase().includes(searchQuery.toLowerCase());
+    });
+  }
+
+  /**
+   * Filter search results when enter is pressed
+   * @param e button click event
+   */
+  function handleEnter(e: any) {
```
Comment (tedspare) [inline frontend/src/SearchMenu/SearchMenu.tsx:98]: nice, this solves an undocumented bug that was causing the page to refresh on enter

File: frontend/src/SearchMenu/SearchMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+              key={tour.id}
+              handleMuralClick={props.handleMuralClick}
+            />
+          );
+        })}
+        {displayedCollections.map((colletion: any) => {
+          return (
+            <SearchResultCard
+              type={FORM.COLLECTION}
+              item={colletion}
```
Comment (tedspare) [inline frontend/src/SearchMenu/SearchMenu.tsx:154]: `colletion` -> `collection`

File: frontend/src/App/App.tsx [hunk 4]
```diff
@@ -88,6 +84,15 @@ function App() {
     setMurals(data.rows);
   };
 
+  /**
+   * Zooms the map to the coordiantes of the clicked searched mural
+   * @param long longitude
+   * @param lat latitude
+   */
+  const handleSearchedMuralZoom = (long: number, lat: number) => {
+    mapRef.current.setLongLat(long, lat);
```
Comment (tedspare) [inline frontend/src/App/App.tsx:93]: another option is to use react-map-gl's FlyToInterpolator method ([docs](https://github.com/visgl/react-map-gl/blob/master/docs/advanced/viewport-transition.md#example-fly-to-a-new-location))

File: frontend/src/SearchResultCard/SearchResultCard.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import { createStyles, makeStyles, Theme } from "@material-ui/core";
+import React from "react";
+import { FORM } from "constants/constants";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    card: {
+      margin: theme.spacing(0, 2, 2, 3),
+      display: "flex",
+      justifyContent: "flex-start",
```
Comment (tedspare) [inline frontend/src/SearchResultCard/SearchResultCard.tsx:12]: could add `width: 90%` or something

Screen Shot 2021-03-10 at 6 10 20 PM

File: frontend/src/App/App.tsx (hunk 4 above)
Comment (JDziewonski98) [reply frontend/src/App/App.tsx:93]: Would it also make sense to close the search menu once you click a mural?

Comment (tedspare) [reply frontend/src/App/App.tsx:93]: yep 100%

File: frontend/src/Map/Map.tsx [hunk 6]
```diff
@@ -33,22 +40,46 @@ function Map({ muralClick, murals }: IMapProps) {
...
+    setLongLat(long: number, lat: number) {
+      setViewport({
+        width: "100vw",
+        height: "100vh",
+        latitude: lat,
+        longitude: long,
+        zoom: PINPOINT_ZOOM,
+        transitionInterpolator: new FlyToInterpolator(),
+        transitionEasing: easeCubic,
+        transitionDuration: 3000,
```
Comment (tedspare) [inline frontend/src/Map/Map.tsx:72]: nice 🏄‍♂️

File: frontend/src/SearchResultCard/SearchResultCard.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  const styles = useStyles();
+
+  function generateBody(): String {
+    switch (props.type) {
+      case FORM.TOUR:
+        return props.item.description
+      case FORM.COLLECTION:
+          return props.item.description
+      case FORM.MURAL:
+        return props.item.address + ", " + props.item.city
```
Comment (tedspare) [inline frontend/src/SearchResultCard/SearchResultCard.tsx:49]: why not `description` for a mural as well?

**Review by tedspare — APPROVED** [review]
LGTM so far

**Pull request merged**